File: duplicate-assert/python-unittest/JavadocTagInfoTest_testSerial.py
import unittest

class JavadocTagInfoTest(unittest.TestCase):

    def test_serial(self):
        ast = DetailAstImpl()
        ast_parent = DetailAstImpl()
        ast_parent.set_type(TokenTypes.LITERAL_CATCH)
        set_parent = ast.__class__.__dict__["set_parent"]
        set_parent.__set__(ast, ast_parent)

        valid_types = [
            TokenTypes.VARIABLE_DEF,
        ]
        for type_ in valid_types:
            ast.set_type(type_)
            self.assertTrue(JavadocTagInfo.SERIAL.is_valid_on(ast),
                            f"Invalid ast type for current tag: {ast.get_type()}")

        ast_parent.set_type(TokenTypes.SLIST)
        ast.set_type(TokenTypes.VARIABLE_DEF)
        self.assertFalse(JavadocTagInfo.SERIAL.is_valid_on(ast),
                         "Should return false when ast type is invalid for current tag")

        ast.set_type(TokenTypes.PARAMETER_DEF)
        self.assertFalse(JavadocTagInfo.SERIAL.is_valid_on(ast),
                         "Should return false when ast type is invalid for current tag")

File: duplicate-assert/java-junit/JavadocTagInfoTest_testSerial.java
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

public class JavadocTagInfoTest {

    @Test
    public void testSerial() throws ReflectiveOperationException {
        final DetailAstImpl ast = new DetailAstImpl();
        final DetailAstImpl astParent = new DetailAstImpl();
        astParent.setType(TokenTypes.LITERAL_CATCH);
        final Method setParent = ast.getClass().getDeclaredMethod("setParent", DetailAstImpl.class);
        setParent.setAccessible(true);
        setParent.invoke(ast, astParent);

        final int[] validTypes = {
            TokenTypes.VARIABLE_DEF,
        };
        for (int type: validTypes) {
            ast.setType(type);
            assertTrue(JavadocTagInfo.SERIAL.isValidOn(ast),
                    "Invalid ast type for current tag: " + ast.getType());
        }

        astParent.setType(TokenTypes.SLIST);
        ast.setType(TokenTypes.VARIABLE_DEF);
        assertFalse(JavadocTagInfo.SERIAL.isValidOn(ast),
                "Should return false when ast type is invalid for current tag");

        ast.setType(TokenTypes.PARAMETER_DEF);
        assertFalse(JavadocTagInfo.SERIAL.isValidOn(ast),
                "Should return false when ast type is invalid for current tag");
    }
}
